packages/kbn-scout/src/reporting/jest.ts

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import path from 'node:path';
+import { Circus, Config } from '@jest/types';
+import { AggregatedResult, BaseReporter, Test, TestCaseResult, TestContext } from '@jest/reporters';
+import { ToolingLog } from '@kbn/tooling-log';
+import { generateTestRunId, getTestIDForTitle, ScoutReport, ScoutReportEventAction } from '.';
+import { environmentMetadata } from '../datasources';
+import { SCOUT_REPORT_OUTPUT_ROOT } from '../paths';
+
+/**
+ * Configuration options for the AppEx QA Jest reporter
+ */
+export interface ScoutJestReporterOptions {
+  name?: string;
+  outputPath?: string;
+}
+
+/**
+ * AppEx QA Jest reporter
+ */
+export class ScoutJestReporter extends BaseReporter {
+  readonly logger: ToolingLog;
+  readonly name: string;
+  readonly runId: string;
+  private report: ScoutReport;
+
+  constructor(
+    jestGlobalConfig: Config.GlobalConfig,
```

Comment:
So with this scout === playwright + jest correct?

Reply:
Do you mind elaborating a bit? I don't understand what you're trying to say.

Reply:
> looks like scout is going to be playwright + jest + mocha right?

This PR is about **reporting** events from a test execution to be used later on for analysis over a longer term. The Mocha and Jest reporters have been added to allow collecting events in the same event-based format from FTR (which is Mocha-based) and the Jest-based test systems in this repo.

Scout is currently handling UI tests using Playwright. There's no support for API tests yet, and the changes in this PR don't have any influence over what test framework we'll eventually be using as a base for API testing.

Reply:
😌